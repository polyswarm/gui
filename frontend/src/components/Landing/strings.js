const strings = {
  title: 'PolySwarm Relay',
  go: 'Go!'
};

export default strings;