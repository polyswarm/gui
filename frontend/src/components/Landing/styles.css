.Landing {
    background-color: #8500ff;
    display: inline-block;
    height: 100vh;
    left: 0;
    top: 0;
    width: 100vw;
}

.Landing-Content {
    color: white;
    display: flex;
    flex-direction: column;
    height: 50%;
    justify-content: flex-end;
    margin: 0 auto;
    width: 40%;
}

.Landing-Content > h2 {
    text-align: center;
}

.Landing-Chain {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
}

.Landing-Nectar {
    display: flex;
    flex-direction: row;
}

.Landing-Nectar > .NectarField {
    flex-grow: 1
}

.Landing-Nectar > .Button {
    margin-top: auto;
    margin-left: 8px;
    height: 48px;
}