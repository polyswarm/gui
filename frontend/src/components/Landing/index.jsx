// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {CSSTransition} from 'react-transition-group';
// Project imports
import NectarField from '../NectarField';
import Button from '../Button';
// Component imports
import strings from  './strings.js';
import './styles.css';

class Landing extends Component {
  constructor(props) {
    super(props);
    this.state = {
        nectar: '',
        error: false,
        hide: false
    };

    this.onNectarChanged = this.onNectarChanged.bind(this);
    this.onButtonClick = this.onButtonClick.bind(this);
  }

  render() {
    const {state: {error, nectar, hide, finished} } = this;
    return(
      <div className='Landing'>
        <div className='Landing-Background'/>
        <div className='Landing-Content'>
          <h1>{strings.title}</h1>
          <div className='Landing-nectar'>
            <CSSTransition
              in={!hide}
              classNames='translate'
              onExited={() => {
                const {props: {onSetNectar}, state: {nectar}} = this;
                if (onSetNectar) {
                  onSetNectar(nectar);
                }
              }}
              timeout={300}>
              {() => (
                <NectarField onChange={this.onNectarChanged}
                  nectar={nectar} />
              )}
            </CSSTransition>
            <Button
              disabled={error || nectar.length == 0}
              onClick={this.onButtonClick} >
              {strings.go}
            </Button>
          </div>
        </div>
      </div>
    );
  }

  onNectarChanged(nectar, valid) {
    this.setState({nectar: nectar, error: !valid});
  }

  onButtonClick() {
    this.setState({hide: true});
  }

}
Landing.proptypes = {
  onSetNectar: PropTypes.func
}
export default Landing;