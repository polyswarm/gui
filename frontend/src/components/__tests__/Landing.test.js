import React from 'react';
import {render, shallow, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Landing from '../Landing';

it('renders without crashing', () => {
  const wrapper = render(<Landing />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('calls onAddressChanged when AddressField is changed', () => {
  const onAddressChanged = jest.spyOn(Landing.prototype, 'onAddressChanged');
  const wrapper = mount(<Landing />);
  wrapper.find('.AddressField').find('input').simulate('change', {target: {value: 'a'}});

  expect(onAddressChanged).toHaveBeenCalledWith('a', false);
});

it('disables the Go button when 0 < length < requirements in AddressField ', () => {
  const wrapper = mount(<Landing />);
  wrapper.find('.AddressField').find('input').simulate('change', {target: {value: 'a'}});

  expect(wrapper.find('.Button').props().disabled).toBeTruthy();
});

it('enables the Go button when address meets length requirements in AddressField', () => {
  const wrapper = mount(<Landing />);
  wrapper.find('.AddressField').find('input').simulate('change', {target: {value: '012345678912345'}});

  expect(wrapper.find('.Button').props().disabled).toBeFalsy();
});

it('disables the Go button when address length is 0', () => {
  const wrapper = mount(<Landing />);
  wrapper.find('.AddressField').find('input').simulate('change', {target: {value: ''}});

  expect(wrapper.find('.Button').props().disabled).toBeTruthy();
});

it('calls setstate with the address when address modified', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing />);

  wrapper.find('.AddressField').find('input').simulate('change', {target: {value: '012345678912345'}});

  expect(setState).toHaveBeenCalledWith({address: '012345678912345', error: false});
});

it('calls setstate with the address & error true when address modified with invalid value', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing />);

  wrapper.find('.AddressField').find('input').simulate('change', {target: {value: '012345912345'}});

  expect(setState).toHaveBeenCalledWith({address: '012345912345', error: true});
});

it('sets hide to true when button is clicked', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing />);
  wrapper.setState({address: '0123456789123456'});
  setState.mockClear();
  wrapper.find('button').simulate('click');

  expect(setState).toHaveBeenCalledWith({hide: true});
});

it('should call onSetAddress when transition finishes', () => {
  const onSetAddress = jest.fn();
  const wrapper = mount(<Landing onSetAddress={onSetAddress} />);
  wrapper.setState({address: '0123456789123456'});

  const instance = wrapper.find('CSSTransition').first().instance();
  instance.onExited();

  expect(onSetAddress).toHaveBeenCalledWith('0123456789123456');
});