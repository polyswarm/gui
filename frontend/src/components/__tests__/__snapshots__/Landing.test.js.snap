// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Landing"
>
  <header
    class="Header"
  >
    <h2>
      PolySwarm Relay
    </h2>
  </header>
  <div
    class="Landing-Content"
  >
    <h2>
      
    </h2>
    <div
      class="Landing-Chain"
    >
      <div
        class="ChainInfo"
      >
        <div
          class="ChainInfo-Title"
        >
          <h2>
            Before Transfer
          </h2>
        </div>
        <div
          class="ChainInfo-Content"
        >
          <p>
            
          </p>
          <ul>
            <li>
              <b>
                asdf
              </b>
              : 1
            </li>
            <li>
              <b>
                asdf
              </b>
              : 1
            </li>
          </ul>
        </div>
      </div>
      <div
        class="ChainInfo"
      >
        <div
          class="ChainInfo-Title"
        >
          <h2>
            After Transfer
          </h2>
        </div>
        <div
          class="ChainInfo-Content"
        >
          <p>
            
          </p>
          <ul>
            <li>
              <b>
                asdf
              </b>
              : 0.5
            </li>
            <li>
              <b>
                asdf
              </b>
              : 1.5
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="Landing-Nectar"
    >
      <div
        class="Dropdown"
      >
        <p
          class="Dropdown-Selection"
        >
          Deposit
        </p>
        <div
          class="Dropdown-Choices"
        >
          <p>
            Deposit
          </p>
          <p>
            Withdraw
          </p>
        </div>
      </div>
      <div
        class="NectarField"
      >
        <label
          class="NectarField-Label"
          for="nectar"
        >
          Amount of Nectar
        </label>
        <input
          class="NectarField-Input"
          id="nectar"
          type="number"
          value=""
        >
          
        </input>
      </div>
      <button
        class="Button"
      >
        Transfer!
      </button>
    </div>
  </div>
</div>
`;
